modules/chataigpt.py

```diff
@@ -0,0 +1,90 @@
+import time
+import speech_recognition as sr
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from gtts import gTTS
+import os
+from bs4 import BeautifulSoup
+import yaml
+
+
+class ChataiGPT:
+    def __init__(self, config_path="chatgpt.yml"):
+        self.load_config(config_path)
+        self.recognizer = sr.Recognizer()
+
+    def load_config(self, config_path):
+        with open(config_path, "r") as config_file:
+            self.config = yaml.safe_load(config_file)
+
+    def check_internet_connection(self):
+        try:
+            driver = webdriver.Chrome()
+            driver.get("https://www.google.com")
```

Comment:
I recommend `http://www.gstatic.com/generate_204` for healthcheck instead of `google.com`. For multiple reasons, but mainly being lighter to load an empty page that returns 204 than the entire google front page. Also note, generate_204 also works great for detecting Captive Portals!